docs/lint-vs-format.md

```diff
@@ -0,0 +1,87 @@
+# Linting & Formatting
+
+## Separation of Concerns
+
+Linting and formatting has been separated in this monorepo. Here are two
+resources that explain why:
+
+- [ESLint is deprecating formatting rules][1]
+- [What About Formatting?][2]
+- [Formatters, linters, and compilers: Oh my!][3] says:
+
+> Don't use linters for formatting!
+>
+> Although lint rules can enforce some formatting preferences in code, they're
+> slower, less consistent, and more prone to missing edge cases when compared to
+> dedicated formatters.
+
+Another reason is that it should be possible to replace either ESLint or
+Prettier without them being too intermingled.
+
+This separation of concerns leads to clearer objectives and faster execution.
+For example, if you want to know whether your code is formatted properly you can
+do `pnpm lint` and get your answer faster compared to doing everything at once.
+
+With that in mind, this document describes some main aspects of the current
+setup.
+
+## Prettier inside ESLint
+
+What about `eslint-plugin-prettier` and `eslint-config-prettier`?
+
+Confusingly perhaps, `eslint-plugin-prettier` loads and runs Prettier inside
+ESLint, while `eslint-config-prettier` disables rules from core and other
+plugins. The latter is what we want and the direction ESLint itself is going.
+
+That's why we've removed `eslint-plugin-prettier`, and include only
+`eslint-config-prettier`.
+
+## Scripts
+
+_tl/dr; Use `pnpm format` for formatting and `pnpm lint` for linting._
+
+When looking into the related scripts in `package.json` you'll find something
+like this:
+
+```json
+{
+  "name": "@kadena/pkg",
+  "scripts": {
+    "format": "pnpm run --sequential /^format:.*/",
+    "format:lint": "pnpm run lint:src --fix",
```

Comment:
Just popping in un-asked-for: I personally have always found it confusing when commands mix "format" and "lint". A lot of devs already have a hard time disambiguating between the two. I'd suggest either calling this `lint:src:fix` or just omitting it (because all it does is add `--fix`.

Reply:
Thanks for popping in, it's appreciated. Naming is hard 🥲

- Renaming to `lint:src:fix` means losing the `format:*` convention, and with that the `/^format:.*/`, in turn resulting in having to manually keep track of them in each package separately.
- Omitting it would mean duplication of something that could change, and lead to subtle bugs as the other script includes two arguments that might change, namely the dir(s) to scan and the extensions (which unfortunately can't be configured elsewhere, like in ESLint config files). They're different across the workspaces. And change rarely, but still.

Since developers only really need to remember `format` and `lint` I thought the existing "mechanical" solution is easier/safer to maintain long-term. Still happy to be convinced otherwise, but these were my considerations.